operator/v1/types_ingress.go

```diff
@@ -225,6 +241,65 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// Ciphers defines the cipher algorithms used by an IngressController for performing
+// encryption or decryption of network connections. A profile can be specified or a
+// custom strategy of ciphers.
```

Comment:
This is either-or (specifying both is not allowed), right?

Reply:
Yes. Even with the latest changes, `IngressController` connection security can be configured  using a `Profile` or `CustomSettings`.

Reply:
A statement such as the following would be appropriate: "Only one of `Profile` and `CustomSettings` should be set."

Reply:
Alternatively, we could make `Profile` required (non-pointer, but still defaulting to `Intermediate`), define `SecurityProfileCustomType SecurityProfileType = "Custom"`, and then add a statement to the effect that `customSettings` must be provided if, and only if, `profile` is `Custom`.

Reply:
> A statement such as the following would be appropriate: Only one of profile and customSettings should be set.

I updated to:
// Profile is one of "Old", "Intermediate", "Modern" or "Custom".
// When a profile of type "Old", "Intermediate" or "Modern" is set, the CustomSettings
// field is forbidden.

Reply:
> define SecurityProfileCustomType SecurityProfileType = "Custom"

Done:
```
	// SecurityProfileCustomType is a security profile that allows for custom settings
	// through type CustomProfileSettings.
	SecurityProfileCustomType SecurityProfileType = "Custom"
```

Reply:
> Alternatively, we could make Profile required (non-pointer, but still defaulting to Intermediate).

I will push a new change set with the `Profile` being required.